samza-core/src/main/scala/org/apache/samza/container/TaskInstance.scala

```diff
@@ -210,7 +207,9 @@ class TaskInstance(
 
     trace("Flushing producers for taskName: %s" format taskName)
 
-    collector.flush
+    exceptionHandler.maybeHandle { // Allow users to decide whether to swallow producer exceptions
```

Comment:
Do we still need this if we're adding the top level task.drop.producer.error config?